front/src/modules/simulationResult/components/SpeedSpaceChart/helpers.ts

```diff
@@ -15,9 +15,71 @@ import type {
 } from 'applications/operationalStudies/types';
 import type { ReportTrainV2 } from 'common/api/osrdEditoastApi';
 import { mmToKm, msToKmh, mToKm } from 'utils/physics';
+import type { ArrayElement } from 'utils/types';
 
 import { electricalProfilesDesignValues } from './consts';
 
+type SpeedLimitTagValue = ArrayElement<SimulationResponseSuccess['mrsp']['values']>;
+const getTag = (source?: SpeedLimitTagValue['source']): { name: string; color: string } => {
+  let name: string;
+  let color: string;
+
+  if (!source) {
+    name = 'missing_from_train';
+    color = '#94918E';
```

Comment:
Instead of "magin colors", do you think you could add an object inside consts.ts like this : 

```suggestion
const TAG_COLORS = {
  MISSING: '#94918E',
  ... (same for other tags)
}
```